internal/resourcestore/resourcecleaner.go

```diff
@@ -1,33 +1,77 @@
 package resourcestore
 
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/cri-o/cri-o/internal/log"
+)
+
 // A CleanupFunc is a function that cleans up one piece of
 // the associated resource.
-type CleanupFunc func()
+type CleanupFunc func() error
 
 // ResourceCleaner is a structure that tracks
 // how to cleanup a resource.
 // CleanupFuncs can be added to it, and it can be told to
 // Cleanup the resource
 type ResourceCleaner struct {
+	ctx   context.Context
 	funcs []CleanupFunc
 }
 
 // NewResourceCleaner creates a new ResourceCleaner
-func NewResourceCleaner() *ResourceCleaner {
+func NewResourceCleaner(ctx context.Context) *ResourceCleaner {
 	return &ResourceCleaner{
+		ctx:   ctx,
```

Comment:
I don't think we're supposed to save ctx as a member variable

Reply:
The main reason is to be able to use `log.Errof` in `Cleanup()`. Do you think we should pass the context on `Cleanup(ctx)` as alternative?

Reply:
Yes, I think it's better to pass `ctx` to `Cleanup` directly.